gloo/allreduce.cc

```diff
@@ -299,7 +307,7 @@ void ring(
     // iterations. At that point we have already sent all data we
```

Comment:
I think it is final `numSegmentsPerRank` iterations instead of final 2 now.

Reply:
It's still final 2, but it reads a bit weird with the condition that includes the `numSegmentsPerRank` subtraction.